### ISSUE TITLE
Seems it does not work for Release with GIT in Bitbucket

### ISSUE THREAD

**Issue by irriss** [rfennell/AzurePipelines#1003]
### Azure DevOps Extensions

Generate Release Notes (Legacy PowerShell)

### Platform

Azure DevOps Services

### Extension Version

3.51.2

### Describe the bug

```bash
I have Azure DevOps build and release pipelines for source code in Bitbucket. I could not make ReleseNotes generator working. So I started investigation and debugging and found out that it works for the build, but doesnt work for Release with comparison with the latest deployment.

I used the tool to narrow down the problem. Method 

// Fall back to original behavior
commits = await buildApi.getChangesBetweenBuilds(artifactInThisRelease.sourceId, parseInt(artifactInMostRecentRelease.buildId),  parseInt(artifactInThisRelease.buildId), 5000);

always returns 0 commits.
```

### Repo Steps

```bash
I found a two releases with difference in a few commits.
Then I run the tool in debug mode using 'overrideBuildReleaseId' parameter. In debug mode I could confirm that parameters artifactInMostRecentRelease.buildId and artifactInThisRelease.buildId are correct ones. This is two lines from console:

Checking what commits and workitems have changed from [20210505.5][ID 9607] => [20210506.1] [ID 9637] 
Using workaround for build API limitation (see issue #349)

I played with Fix349 parameter but it did not help. Seems that the issue is in buildApi.getChangesBetweenBuilds method.
```

### Expected Behavior

```bash
Method should return a collection of changesets but returns empty collection
```

**Comment by rfennell** [comment]
Given the version number I think you are using the cross platform node based version of my Release Notes.

I will have to spin up a Bitbucket instance to retest this, but my initial questions are

- Is the bitbucket instance cloud hosted or locally hosted?
- Is the bitbucket project public or private?

**Comment by irriss** [comment]
Thanks for prompt reply, Richard!

>Given the version number I think you are using the cross platform node based version of my Release Notes.
that's correct

>Is the bitbucket instance cloud hosted or locally hosted?

cloud

>Is the bitbucket project public or private?

private

**Comment by rfennell** [comment]
I have re-tested and agree that when the source repository is Bitbucket that there is an issue

| Usage | API call | Result |
|-|-|-|
| Classic Build | `getBuildChanges` - Get the changes associated with the build since the last successful build | Works|
| Single-stage YAML build | `getBuildChanges` - Get the changes associated with the build since the last successful build | Works|
| Classic Release first run | `getBuildChanges` - Get the changes associated with the build since the last successful build | Works|
| Multi-Stage YAML Build first run | `getBuildChanges` - Get the changes associated with the build since the last successful build | Works|
| Classic Release other runs | `getChangesBetweenBuilds` - Get the changes between two builds | Returns Nothing|
| Multi-Stage YAML Build other runs |  `getChangesBetweenBuilds` - Get the changes between two builds | Returns Nothing|

The issue is with the call 'getChangesBetweenBuilds' in the Microsoft Node SDK. I have checked that the direct [REST API call](https://docs.microsoft.com/en-us/rest/api/azure/devops/build/builds/get%20changes%20between%20builds?view=azure-devops-rest-6.1) exhibits the same results i.e. nothing returned.

This is a limitation of the Microsoft provided REST API, nothing I can really do here.

I have logged an issue [on the Microsoft project repo](https://github.com/microsoft/azure-devops-node-api/issues/444)

**Comment by ruslan-oncoshot** [comment]
Hi Richard, apparently it does not work with GitHub either.
getChangesBetweenBuilds and getWorkItemsBetweenBuilds do not return anything

**Comment by rfennell** [comment]
That makes sense, Azure DevOps only fully support detail when the source is located internally to Azure DevOps. I will update the [issue if have raised with Microsoft](https://github.com/microsoft/azure-devops-node-api/issues/444)

**Comment by ruslan-oncoshot** [comment]
As far as I understand we can get edge commits from AzureDevOps builds and then use GitHub API to get the difference between them?
https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/GitHubReleaseV0/operations/ChangeLog.ts

**Comment by rfennell** [comment]
Yes a potential solution, but something I would like to avoid if I can.
